src/App.tsx

```diff
@@ -1,24 +1,195 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useCallback, useEffect, useState } from 'react';
+import { TodoInput } from './components/TodoInput';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Errors } from './types/Errors';
+import * as todosApiServices from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodosCounter } from './components/TodosCounter';
+import { ClearCompleted } from './components/ClearCompleted';
+import { ToggleAll } from './components/ToggleAll';
 
-const USER_ID = 0;
+export const USER_ID = 12146;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.all);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [loadingTodosPause, setLoadingTodosPause] = useState<number[]>([]);
+
+  const hasCompletedTodos
+    = todos?.some(({ completed }) => completed === true);
+
+  useEffect(() => {
+    setErrorMessage(null);
+  }, [filter, todos]);
+
+  function loadPosts() {
+    setLoading(true);
+
+    todosApiServices.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMessage(Errors.UnableToLoadTodos))
+      .finally(() => setLoading(false));
+  }
+
+  useEffect(loadPosts, []);
+
+  const addTodo = useCallback((title: string): Promise<void> => {
+    setLoadingTodosPause(prev => [...prev, 0]);
+    const newTodo = {
+      userId: USER_ID,
+      title,
+      completed: false,
+      id: 0,
+    };
+
+    setTempTodo({ ...newTodo });
+
+    return todosApiServices.addTodo(newTodo)
+      .then((response) => setTodos(
+        currentTodos => [...currentTodos, response],
+      ))
+      .then(() => setTempTodo(null))
+      .catch(() => {
+        setErrorMessage(Errors.UnableToAddATodo);
+        throw new Error(Errors.UnableToAddATodo);
+      })
+      .finally(
+        () => setLoadingTodosPause(prevIds => prevIds
+          .filter(prevId => prevId !== 0)),
+      );
+  }, [errorMessage]);
```

Comment:
![2024-01-08_16h36_06](https://github.com/mate-academy/react_todo-app-with-api/assets/61904995/65c62c34-f827-4e65-a527-87e1780e4a09)
```suggestion
  }, []);
```